src/Symfony/Bridge/PhpUnit/Legacy/CoverageListenerTrait.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\PhpUnit\Legacy;
+
+use PHPUnit\Framework\Test;
+
+/**
+ * PHP 5.3 compatible trait-like shared implementation.
+ *
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ *
+ * @internal
+ */
+class CoverageListenerTrait
+{
+    private $sutFqcnResolver;
+
+    public function __construct(callable $sutFqcnResolver = null)
+    {
+        $this->sutFqcnResolver = $sutFqcnResolver;
+    }
+
+    public function startTest($test)
+    {
+        $annotations = $test->getAnnotations();
+
+        if (isset($annotations['class']['covers']) || isset($annotations['method']['covers'])) {
+            return;
+        }
+
+        $sutFqcn = $this->findSutFqcn($test);
+        if (!$sutFqcn) {
+            return;
```

Comment:
IMHO this is not a good idea. Because you may have some Integration / Functional / Acceptance tests. And theses test are not related to another class. It's just a place where to put some test.
So triggering a warning each time will be really boring for the end user.